docs/development/abi.md

```diff
@@ -0,0 +1,219 @@
+# Pyodide Platform ABI
+
+## ABIs
+
+### General
+
+Shared libraries must be linked with `-sSIDE_MODULE=1` or
+
+```
+-sSIDE_MODULE=2 -sEXPORTED_FUNCTIONS=<export list>
+```
+
+Note that the name of each symbol in the list needs to be prefixed with an
+underscode. For the smallest result, it is recommended to link with:
+
+```
+-sSIDE_MODULE=2 -sEXPORTED_FUNCTIONS=["_PyInit_MyCModule1", "_PyInit_MyCModule2]
+```
+
+To force all symbols to be exported, link with `-sSIDE_MODULE=1`.
+
+If `-pthread` is used at compile or link time, the resulting libraries will not
+load.
```

Comment:
In terms of a spec - should this use RFC2119 language? i.e., "`-pthread` MUST NOT be used in the compilation or linking of a module.

Reply:
I like the style where we explain the consequences of particular actions rather than dictating what people do. People are free to build libraries with `-pthread`. They just won't work. I'm trying to explain here what works.

But if you have a strong position in favor of RFC2119, I would consider wording stuff that way.

Reply:
No objection to clarifying the consequences (that modules won't load); but adhering to the generally accepted language for a formal specification can only be beneficial IMHO. I'd suggest:

"`-pthread` must not be used at compile or link time. If `-pthread` is used, the resulting libraries will not load.